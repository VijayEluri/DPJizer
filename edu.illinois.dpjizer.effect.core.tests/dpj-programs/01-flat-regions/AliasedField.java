/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package flatregion.aliased;

/**
 * 
 * @author Mohsen Vakilian
 * @author Danny Dig
 *
 */
class MyPair {
	int x;
	int y;
	
	MyPair() {}
}

class AliasedField {
	region r1;
	MyPair p in r1;
	
	AliasedField() {}
	
	void m() {
		MyPair localP = p;
		localP.x = 3;
	}
}
