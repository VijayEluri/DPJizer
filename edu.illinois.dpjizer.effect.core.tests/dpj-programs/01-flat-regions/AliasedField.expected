flatregion.aliased.AliasedField.AliasedField() invokes java.lang.Object.Object()
flatregion.aliased.AliasedField.m() reads Root : flatregion.aliased.AliasedField.r1 writes Root
flatregion.aliased.MyPair.MyPair() invokes java.lang.Object.Object()
